apps/andi/lib/andi_web/controllers/api/ingestion_controller.ex

```diff
@@ -144,9 +144,23 @@ defmodule AndiWeb.API.IngestionController do
   end
 
   defp add_uuid(message) do
-    uuid = UUID.uuid4()
-
-    Map.merge(message, %{"id" => uuid}, fn _k, v1, _v2 -> v1 end)
+    if Map.has_key?(message, "id") do
+      case IngestionStore.get(Map.get(message, "id")) do
+        {:ok, nil} ->
+          {:invalid, "Do not include id in create call"}
```

Comment:
Nice

Reply:
Ohh okay. Took me a minute to parse this function. "ifIngestionAlreadyExists" or "ifNewIngestion" would have gone a long way for me to understand I think. Is this check so that the ID can be determined differently, depnding on if this is an Update operation, or a create operation? Someway to indicate that more literally in the code might be nice, but I think the logic here is sound. Does a nice job compromising for the fact that atm, creating / updating happen on the same REST endpoint.

Reply:
Yup yup. And I wasn't sure the most elixir-idiomatic way of doing this. Just kind of threw some matchers at it, but I could see something like 

`if caseExists? do...`

and 
```
def caseExists? do
    case IngestionStore.get(Map.get(message, "id")) do
        {:ok, nil} -> false
        ...
```